File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03086/p03086_num5_parsed.py
def longest_acgt_substring(S):
    max_length = 0
    current_length = 0

    for char in S:
        if char in {'A', 'C', 'G', 'T'}:
            current_length += 1
        else:
            max_length = max(max_length, current_length)
            current_length = 0

    return max(max_length, current_length)

S = input()
print(longest_acgt_substring(S))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03086/p03086_num5_parsed.cpp
#include<iostream>
#include<string>
using namespace std;

int longestACGT(string S) {
    int maxLength = 0;
    int currLength = 0;

    for (int i = 0; i < S.length(); i++) {
        if (S[i] == 'A' || S[i] == 'C' || S[i] == 'G' || S[i] == 'T') {
            currLength++;
        } else {
            maxLength = max(maxLength, currLength);
            currLength = 0;
        }
    }

    maxLength = max(maxLength, currLength);

    return maxLength;
}

int main() {
    string S;
    cin >> S;

    int result = longestACGT(S);
    cout << result << endl;

    return 0;
}